lib/types/string.js

```diff
@@ -180,6 +180,13 @@ class StringPrompt extends Prompt {
     this.write([header, prompt, footer].filter(Boolean).join('\n'));
     this.restore();
   }
+
+  set value(value) {
+    this.state.value = value;
+  }
+  get value() {
+    return [this.state.value, this.input].find(this.isValue.bind(this)) || this.initial;
+  }
```

Comment:
Hmm, shouldn't this be fixed [here](https://github.com/enquirer/enquirer/blob/master/lib/prompt.js#L361) instead?

Reply:
> That will influence other prompts like boolean prompt. (Of course I have tested.)

Right, that's the reason it's in the base `Prompt` class. Ideally, the logic should be simple enough that it works for every prompt. Of course, we can override it as you did in your PR if necessary. But first I want to try something... IMHO the code in your PR makes more sense than what's currently defined in `Prompt`. Before we merge in your changes, would you mind trying your code again in the `Prompt` class, but with the following code in the boolean prompt (and remove the code you added to the string prompt)?

```js
  set value(value) {
    super.value = value;
  }
  get value() {
    return this.cast(super.value);
  }
```

This way we're still using the `value` logic in the `Prompt` class, and if that changes this should still work.

Reply:
thanks for doing this, btw!

Reply:
Updated.